include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -387,8 +389,9 @@ inline void multiplexer::do_start_receive()
 inline void multiplexer::discard_message() {
     boost::system::error_code error;
     endpoint_type remote_endpoint;
+    static char empty_buffer_storage;
     next_layer().receive_from
-        ( boost::asio::buffer(static_cast<char*>(nullptr), 0)
+        ( boost::asio::buffer(&empty_buffer_storage, 1)
```

Comment:
> As I mentioned by email, I think the fact that zero length UDP datagrams work at all on any platform is undefined behaviour

Oh, I thought you meant it's undefined behavior due to Asio not documenting it. I see now that you meant that it's undefined behavior because platforms don't explicitly document it. Makes sense.

> Maybe "nicer" support would be an explicit empty message "poke the ASIO reactor" datagram?

I don't mind, but we should probably read as little as possible. Also, I still think a dedicated method that returns the buffer would be nice (if only to have fewer statics in the code).